 Дано, что орёл выпал на 10% больше, чем решка, всего было $n$ бросков. Считаем, что выпадение орла - это успех, посчитаем долю получившихся успехов от n. Пусть выпало $x$ решек, тогда по условию задачи выпало $1.1\cdot{x}$ орлов. Тогда всего было $2.1\cdot{x}$ бросков, что равняется $n$. Найдём долю выпавших орлов от всех бросков: кол-во орлов = $\frac{n}{2.1}\cdot{1.1}$ = $0.52381\cdot{n}$

Далее нам нужно применить биномиальный тест для проверки нашей задачи: https://en.wikipedia.org/wiki/Binomial_test

Броски монеты - это случайная величина, которая распределена биномиально, так как в качестве исходов может быть только успех и не успех, при этом каждый раз они наступают с одинаковой вероятностью, то есть новые исходы не зависят от старых. Поэтому мы будем использовать как раз этот тест, который позволит проверить нам, является ли монета честной, то есть что успешные исходы наступают с вероятностью 0.5.

$H_0: p=0.5;$

$H_1: p!=0.5;$

Мы выдвигаем нулевую гипотезу: вероятность успешных событий равняется 0.5. В качестве альтернативной гипотезы мы рассматриваем, что вероятность не равна 0.5, поэтому тест будет two-sided, так как вероятность может быть искажена в любую из двух сторон.

Будем использовать уровень значимости = 5%, так как его достаточно часто используют в тестах.

С помощью цикла и библиотеки scipy переберём разные значения $n$, проведём стат.тест и проверим, при каком $n$ получим такое критическое значение, что можно будет отклонить нулевую гипотезу.

In [2]:
from scipy import stats

In [30]:
for n in range(2, 1000):
    '''
    Ниже первый параметр - количество успехов
    Второй параметр - количество экспериментов (бросков n)
    Третий параметр - вероятность успеха в нулевой гипотезе
    '''
    p_value = stats.binom_test(0.52381 * n, n=n, p=0.5, alternative='two-sided')
    if p_value < 0.5:
        print("При {} количестве бросков можно отклонить нулевую гипотезу о том,"\
        " что монета честная.".format(n))
        print("p-value равняется {}".format(p_value))
        break

При 252 количестве бросков можно отклонить нулевую гипотезу о том, что монета честная.
p-value равняется 0.4884260182866813


Ответ: $n$ >= 252, при таком количестве бросков можно утверждать, что монета нечестная.